### DISCUSSION TITLE
Adding NEW READING FORMAT

### DISCUSSION THREAD

**Discussion by mbaggnx** [porrey/Virtual-ZPL-Printer#27] [Ideas]
Hello Daniel, how are you?

I have another idea wich would be more interesting to develop inside of the same ZPL reader

Most of the times, not only etiquetes are being sent, even some other type of documents that they are not in ZPL format, they are in kinda "old" format (mostly from old system users)

Do you thhink that it is possible to make a readable printing for a format characters like I atatch?

[ALB TST.txt](https://github.com/porrey/Virtual-ZPL-Printer/files/10741495/ALB.TST.txt)

(actually, the ZPL does not works with this kind of spool)

**Comment by rafamuz** [comment]
I'd suggest it right now. Sometimes, not only RAW format is being sent (with ZPL commands), but other like PDFs and other formats like PNG images, etc... but the virtual printer only reads and processes labels from ZPL formats. the idea here is to use this virtual printer to run exactly as a normal zebra printer does, i.e., when you send a ZPL txt or ZPL by streaming (RAW) it prints, converting the commands to the final label, but when you send PDFs, IMAGES, Word files, etc... from any other program, it also prints directly (as a normal Laser printer). THIS IS WHAT A ZEBRA PRINTER does. It will SAVE our time and money, once this software will, now, work as a ZEBRA converter for those CHINESE (and very cheaper) printers. Here in my country, the price for a ZEBRA printer is 3 to 5 times more expensive than these chinese ones! So. It does not work as I wanted. I use the site PRINT NODE, to receive from my ERP, some labels, sometimes they came by RAW, sometimes they come by PDF format. So once I receive it as ZPL RAW, i can print on my PHYSICAL Chinese printer. But when receiving PDF, I can't set a rule to print it without need to labelary it...So it gets label erros... It's a very good idea!  The idea is SKIP labelary and just show the content on the screen and/or re-sending it to the physical printer. Very nice!!!

![Imagem 001](https://github.com/user-attachments/assets/e8501e00-a82f-44ba-8562-305ac9495dcc)

![Imagem 002](https://github.com/user-attachments/assets/3567b183-5000-4641-aa50-21af0e94e345)

![Imagem 003](https://github.com/user-attachments/assets/3b265187-0840-4f34-ae38-3e0124061a49)

![Imagem 004](https://github.com/user-attachments/assets/607aacbd-8c51-48e8-bbdc-7d1765c12cf4)
